cnf-certification-test/lifecycle/scaling/crd_scaling.go

```diff
@@ -0,0 +1,232 @@
+// Copyright (C) 2020-2022 Red Hat, Inc.
+//
+// This program is free software; you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation; either version 2 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License along
+// with this program; if not, write to the Free Software Foundation, Inc.,
+// 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+
+//nolint:dupl
+
+package scaling
+
+import (
+	"context"
+	"errors"
+	"time"
+
+	"github.com/sirupsen/logrus"
+	"github.com/test-network-function/cnf-certification-test/cnf-certification-test/lifecycle/podsets"
+	"github.com/test-network-function/cnf-certification-test/internal/clientsholder"
+	"github.com/test-network-function/cnf-certification-test/pkg/tnf"
+	scalingv1 "k8s.io/api/autoscaling/v1"
+	v1 "k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1"
+	scale "k8s.io/client-go/scale"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	hps "k8s.io/client-go/kubernetes/typed/autoscaling/v1"
+	retry "k8s.io/client-go/util/retry"
+)
+
+func ScaleAcrTest(aCrd *v1.CustomResourceDefinition, horizontalScaler []*scalingv1.HorizontalPodAutoscaler, timeout time.Duration) []unstructured.Unstructured {
+	clients := clientsholder.GetClientsHolder()
+	var failedcrd []unstructured.Unstructured
+
+	for _, version := range aCrd.Spec.Versions {
+		gvr := schema.GroupVersionResource{
+			Group:    aCrd.Spec.Group,
+			Version:  version.Name,
+			Resource: aCrd.Spec.Names.Plural,
+		}
+
+		logrus.Debugf("Looking for CRs from CRD: %s api version:%s group:%s plural:%s", aCrd.Name, version.Name, aCrd.Spec.Group, aCrd.Spec.Names.Plural)
+		crs, err := clients.DynamicClient.Resource(gvr).List(context.TODO(), metav1.ListOptions{})
+
+		if err != nil {
+			logrus.Errorf("error getting Resource for %s: %v\n", aCrd.Name, err)
+		}
+		for _, cr := range crs.Items {
+			groupResourceSchema := schema.GroupResource{
+				Group:    aCrd.Spec.Group,
+				Resource: aCrd.Spec.Names.Plural,
+			}
+			namespace := cr.Object["metadata"].(map[string]interface{})["namespace"].(string)
```

Comment:
We need to run this test only in the namespaces under test, not on all namespaces that CR are defined on. I would suggest reusing the getCrsPerNamespaces funtion that returns a list of CR names per namespace loop throught the namespaces under test to test all the CR under test.